### PR TITLE
fix: resolve wasm build errors

### PR DESCRIPTION
**Pull request by doinkythederp**
## Describe the changes this PR makes. Why should it be merged?

Fixes WASM build errors by using a custom version of libc_print and moving an unsafe function into an unsafe block.

## Additional Context
- These are breaking changes (semver: major).
- These changes update the crate's interface (e.g. functions/modules added or changed).
- I have tested these changes on a VEX V5 brain.
- I have tested these changes in a simulator.

### CODE REVIEW

File: packages/pros/src/io/print_impl.rs [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+#![warn(unsafe_op_in_unsafe_fn)]
+
+use core::{convert::TryFrom, file, line, stringify};
+
+/// This forces a "C" library linkage
+#[cfg(not(windows))]
+#[link(name = "c")]
+mod c {
+    extern "C" {}
+}
```
Comment (Gavin-Niederman) [inline packages/pros/src/io/print_impl.rs:76]: why is this necessary?

File: packages/pros/src/io/print_impl.rs [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+
+// These constants are used by the macros but we don't want to expose
+// them to library users.
+#[doc(hidden)]
+pub const __NEWLINE: &str = "\n";
+#[doc(hidden)]
+pub const __STDOUT: i32 = 1;
+#[doc(hidden)]
+pub const __STDERR: i32 = 2;
```
Comment (Gavin-Niederman) [inline packages/pros/src/io/print_impl.rs:85]: I don't see why these need to exist at all. Especially the `__NEWLINE` constant.

File: packages/pros/src/io/print_impl.rs [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+
+#[doc(hidden)]
+pub struct __ProsSysWriter(i32);
```
Comment (Gavin-Niederman) [inline packages/pros/src/io/print_impl.rs:88]: I think a better name would be `__SerialWriter` or similar.

File: packages/pros/src/io/print_impl.rs [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+    })
+    .ok()
+}
+
+/// Macro for printing to the standard output, with a newline.
+///
+/// Does not panic on failure to write - instead silently ignores errors.
+///
+/// See [`println!`](https://doc.rust-lang.org/std/macro.println.html) for
+/// full documentation.
```
Comment (Gavin-Niederman) [inline packages/pros/src/io/print_impl.rs:152]: Does everything in the standard library docs actually apply to these macros?

**Review by Gavin-Niederman — CHANGES_REQUESTED** [review]
Some questions I had.

File: packages/pros/src/io/print_impl.rs (hunk 1 above)
Comment (Gavin-Niederman) [reply packages/pros/src/io/print_impl.rs:76]: I don't think its correct here because we don't actually have to link with libc here. These come from libpros.

File: packages/pros/src/io/print_impl.rs (hunk 2 above)
Comment (doinkythederp) [reply packages/pros/src/io/print_impl.rs:73]: This is probably better to ask the original library author about.

Comment (Gavin-Niederman) [reply packages/pros/src/io/print_impl.rs:73]: Can you remove them?

File: packages/pros/src/io/print_impl.rs (hunk 4 above)
Comment (doinkythederp) [reply packages/pros/src/io/print_impl.rs:132]: Presumably, since its using the same formatting API under the hood.

**Review by Gavin-Niederman — APPROVED** [review]
LGTM (Legit Great Terminal Macros)
